### PR TITLE
Fluid manager

### PR DESCRIPTION
**Pull request by burzum**

### CODE REVIEW

**Comment by ADmad** [comment]
Can you please add tests for `Manager::add()` with options and for custom `typeClasses` mapping.

**Comment by burzum** [comment]
@ADmad done

File: tests/TestCase/ManagerTest.php [hunk 1]
```diff
@@ -2,11 +2,17 @@
 namespace Search\Test\TestCase;
 
 use Cake\Core\Configure;
-use Cake\ORM\Entity;
-use Cake\ORM\Table;
 use Cake\ORM\TableRegistry;
 use Cake\TestSuite\TestCase;
 use Search\Manager;
+use Search\Type\base;
```
Comment (ADmad) [inline tests/TestCase/ManagerTest.php]: Typo "base" => "Base"

**Comment by burzum** [comment]
@ADmad fixed.

**Comment by jippi** [comment]
-1 for killing existing method names - at least add them back for compat

**Comment by burzum** [comment]
@jippi -1 for not reviewing the change properly. :stuck_out_tongue: 

```php
    public function __call($method, $args)
     {
        if (!isset($args[1])) {
            $args[1] = [];
        }
        return $this->add($args[0], $method, $args[1]);
     }
```

This will delegate the old method calls.

**Comment by jippi** [comment]
got tests to prove it? :D

**Comment by burzum** [comment]
The existing tests for two of the filters (why are they named types?) still pass. This is prove enough. I'm not going to add the tests you forgot to write initially.

**Comment by jippi** [comment]
:dash:

**Comment by ADmad** [comment]
@burzum Tests are missing for the new `Manager::collection()` method.

**Comment by burzum** [comment]
@ADmad still something else...?

**Comment by ADmad** [comment]
@burzum Looks good to me. Just one more thing. Even though the magic `__call()` takes of old methods it would be better to keep those concrete methods and make them use the new `add()` method.

The reasoning is similar to what @lorenzo stated in cakephp#7684. Having concrete method makes the API more readable and available methods easier to find.

**Comment by burzum** [comment]
@ADmad and now...?

**Pull request merged**